.github/workflows/asv-pr.yml

```diff
@@ -0,0 +1,103 @@
+# This workflow will run benchmarks with airspeed velocity (asv) for pull requests.
+# It will compare the performance of the main branch with the performance of the merge
+# with the new changes and publish a comment with this assessment.  
+
+name: Run ASV benchmarks for PR
+
+on:
+  pull_request:
+    branches: [ main ]
+
+env:
+  PYTHON_VERSION: "3.10"
+  WORKING_DIR: ${{ github.workspace }}/benchmarks
+
+jobs:
+
+  setup-python:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Cache Python ${{ env.PYTHON_VERSION }}
+        uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: python-${{ env.PYTHON_VERSION }}
+
+      - name: Set up Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: "${{ env.PYTHON_VERSION }}"
+
+  asv-pr:
+    runs-on: ubuntu-latest
+    needs: setup-python
+
+    permissions:
+      actions: read
+      pull-requests: write
+
+    defaults:
+      run:
+        working-directory: ${{ env.WORKING_DIR }}
+
+    steps:
+      - name: Checkout PR branch of the repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Cache Python ${{ env.PYTHON_VERSION }}
```

Comment:
What are the differences between this and the previous cache step on line 21?

Reply:
I don't fully know. These workflows were copied over from within the python-project-template. Sandro might have some more insight here (but I can't tag him because he's not a member of the organization yet). It looks like it's duplicated in asv-pr and asv-main, but not asv-nightly. Not sure if that's relevant?

Reply:
Opened https://github.com/lincc-frameworks/python-project-template/issues/259